src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3]; // For example "/"
+  const newSeparator = toFormat[3]; // For example "."
+  const newFormat = toFormat;
+  const newFormatSorted = [...toFormat].sort(); // for easy found YYYY or YY ??
+  let YearFormFromFormat = ''; // found what we in old format YY OR YYYY ??
+  const arrayFromDataString = date.split(oldSeparator);
+  const objectOfDataAndFormats = {};
+  // object =  keys = index of fromFormat, values = digits of date;
+  const fromFormatLenghtWitoutSeparator = fromFormat.length - 1;
```

Comment:
There is a typo in the variable name `fromFormatLenghtWitoutSeparator`. It should be `fromFormatLengthWithoutSeparator`.